agent/pkg/singularity/singularity.go

```diff
@@ -387,6 +378,51 @@ func (s *SingularityClient) RunContainer(
 	}, nil
 }
 
+// Sets the environment of the process that will run the Singularity/apptainer command.
+func (s *SingularityClient) setCommandEnvironment(
+	req cproto.RunSpec, cudaVisibleDevices []string, cmd *exec.Cmd,
+) {
+	// Per https://pkg.go.dev/os/exec#Cmd.Env, if cmd.Env is nil, the new process uses the current
+	// process's environment. If this in not the case, for example because we specify something to
+	// control Singularity operation, then we need to explicitly specify any value needed from the
+	// current environment.
+	if req.DeviceType == device.CUDA {
+		cudaVisibleDevicesVar := strings.Join(cudaVisibleDevices, ",")
+		cmd.Env = append(cmd.Env,
+			fmt.Sprintf("SINGULARITYENV_CUDA_VISIBLE_DEVICES=%s", cudaVisibleDevicesVar),
+			fmt.Sprintf("APPTAINERENV_CUDA_VISIBLE_DEVICES=%s", cudaVisibleDevicesVar),
+		)
+	}
+	if req.DeviceType == device.ROCM {
+		// Avoid this problem: https://github.com/determined-ai/determined-ee/pull/922
+		// by not setting both ROCR_VISIBLE_DEVICES & CUDA_VISIBLE_DEVICES
+		s.addToTargetCommandEnvironmentIfSet(cmd, "ROCR_VISIBLE_DEVICES")
```

Comment:
Interesting that it doesn't need to be 

SINGULARITYENV_ROCR_VISIBLE_DEVICES

???